sec1_intro_and_setup/test_app/config/locales/en.yml

```diff
@@ -0,0 +1,33 @@
+# Files in the config/locales directory are used for internationalization
```

Comment:
Whats this file for?

Reply:
This file is a translational file that ensures what language is used in your application, by default is english language so `en.yml`, if your application is going to have multiple language options you can include all the translations as `yml` files in `config/locales`.

Reply:
1. How do you `include` all translations?
2. How does the translation work?

Reply:
1. Choose the languages that you hope to support in `config/application.rb` and use the `i18n` gem to do the translation (`i18n` has locale data for different languages)
2. Rails will automatically pick the proper view based on your current set locale. However, let say you are in Korea and you still want your app to be used in English, then when developing apps, you would need to include the translations files (`.yml`) in `config/locales` and the apps will render based on the chosen language but not based on locale.

Reply:
1. How? you aren't answering my question. Stop giving me theory and give me code. Real code. By giving me code is when you really know whether you are correct or not. By telling me all these theories which you read from the internet, it doesn't show me whether you really know what you read or not. Plus, how can you be sure what you read is correct?
2. You say it will render based on the chosen language. How does "chosen language" get set?

Reply:
1. ok
2. What if we do not use the http header to set language but pass in as a params instead?